qiskit/pulse/schedule.py

```diff
@@ -926,65 +478,1109 @@ def draw(self,
+        super().__init__(name=name, metadata=metadata)
+
+        self._transform_policy = transform
+        self._transform_opts = transform_opts
+
+        self._blocks = list()
+
+        for block in blocks:
+            self.append(block, inplace=True)
+
+    @property
+    def transform(self) -> str:
+        """Return callback function that align block component to generate schedule."""
+        return self._transform_policy
+
+    @property
+    def transform_opts(self) -> Dict[str, Any]:
+        """Return keyword arguments that is used for transformation."""
+        return self._transform_opts
+
+    def schedule_ready(self) -> bool:
+        """Return ``True`` if all durations are assigned."""
+        return all(block.schedule_ready() if isinstance(block, ScheduleBlock) else
+                   isinstance(block.duration, int) for block in self.instructions)
+
+    @property
+    @_require_schedule_conversion
+    def timeslots(self) -> TimeSlots:
+        """Time keeping attribute."""
+        return self.timeslots
+
+    @property
+    @_require_schedule_conversion
+    def duration(self) -> int:
+        """Duration of this schedule block."""
+        return self.duration
+
+    @property
+    @_require_schedule_conversion
+    def start_time(self) -> int:
+        """Starting time of this schedule block."""
+        return self.ch_start_time(*self.channels)
+
+    @property
+    @_require_schedule_conversion
+    def stop_time(self) -> int:
+        """Stopping time of this schedule block."""
+        return self.duration
+
+    @property
+    def channels(self) -> Tuple[Channel]:
+        """Returns channels that this schedule clock uses."""
+        chans = set()
+        for block in self.instructions:
+            for chan in block.channels:
+                chans.add(chan)
+
+        return tuple(chans)
+
+    @property
+    def instructions(self) -> Tuple[BlockComponent]:
+        """Get the time-ordered instructions from self."""
+        return tuple(self._blocks)
+
+    @_require_schedule_conversion
+    def ch_duration(self, *channels: Channel) -> int:
+        """Return the time of the end of the last instruction over the supplied channels.
+
+        Args:
+            *channels: Channels within ``self`` to include.
+        """
+        return self.ch_duration(*channels)
+
+    @_require_schedule_conversion
+    def ch_start_time(self, *channels: Channel) -> int:
+        """Return the time of the start of the first instruction over the supplied channels.
+
+        Args:
+            *channels: Channels within ``self`` to include.
+        """
+        return self.ch_start_time(*channels)
+
+    @_require_schedule_conversion
+    def ch_stop_time(self, *channels: Channel) -> int:
+        """Return maximum start time over supplied channels.
+
+        Args:
+            *channels: Channels within ``self`` to include.
+        """
+        return self.ch_stop_time(*channels)
+
+    @_method_not_supported
```

Comment:
Because other methods doesn't return schedule object, i.e. when we call `duration`, the converted schedule is just discarded. However, this method returns schedule and object type is implicitly replaced. I think this makes debugging of code quite hard and I'm trying to avoid.

Reply:
Ok sounds good. I added another comment below on a docstring about how we might make it more clear what the plans are moving forward on each of these methods.

I'm thinking shift and insert won't be reached from the builder, so maybe they're nothing to worry about, but `filter` and `exclude` are used pretty frequently and could use some info along the lines of "coming soon!"

Reply:
Actually, I resolved that comment to consolidate threads. This was it:
```
Here, I would say that support will be added soon. Might be nice if the warnings for methods that aren't yet supported (filter, exclude) is different from those that we are actually deprecating (insert, shift?)
```
It was about the docstring note of `filter`